home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/cooperation/apartment/ApartmentServiceImpl.java

```diff
@@ -42,19 +42,23 @@
     @Override
     public ApartmentDto createApartment(Long houseId, ApartmentDto createApartmentDto) {
 
-        var house = houseRepository.findById(houseId)
-                .filter(House::getEnabled)
-                .orElseThrow(() -> new NotFoundHomeException(
-                        String.format(HOUSE_WITH_ID_NOT_FOUND, houseId)));
+        if (apartmentRepository.findByApartmentNumberAndHouseId(createApartmentDto.getApartmentNumber(), houseId)
+            .isPresent()) {
+            throw new BadRequestHomeException("Apartment with number " + createApartmentDto.getApartmentNumber()
```

Comment:
Please add test in `home-api-tests` module that will cover this condition.

Reply:
Done.